pkg/cli/cli.go

```diff
@@ -34,11 +33,21 @@ type IOStreams struct {
 	ErrOut io.Writer
 }
 
+type CommandRunner interface {
+	// Start a command and doesn't wait for it to exit. Instead it reads its entire
+	// standard output and returns that or an error. The commands stdin and stderr
```

Comment:
isn't there a conflict between not waiting for it to exit and reading the entire standard output?

Reply:
A background process (daemon) usually closes its standard file descriptors (https://netzmafia.ee.hm.edu/skripten/unix/linux-daemon-howto.html). It's common for the parent process launching a daemon to ensure that it started correctly and the easiest way to do that is to read from a pipe shared between both processes. The fact that said pipe is the daemon's stdout is a minor detail for us.
BTW this is the same thing `launch_cvd --daemon` does, except that it uses a different fd than stdout IIRC.

Reply:
I see, yeah I asked because cvd_server does not close the file descriptors so the execution gets blocked there https://github.com/google/android-cuttlefish/blob/main/frontend/src/host_orchestrator/orchestrator/instancemanager.go#L577-L582. I wasn't aware that it's expected for good "dameon" citizens to close its standard file descriptors.

Reply:
good daemons actually do `fork()` + `exec()` so you can actually just wait for them to exit and it's that daemon initial process that waits for some signal from the child to exit itself instead of waiting for it...
But that's impossible with Go because `fork()` only recreates the main thread and that breaks the runtime.